src/ray/raylet/local_object_manager.cc

```diff
@@ -231,15 +231,23 @@ void LocalObjectManager::SpillObjectsInternal(
                 num_active_workers_ -= 1;
               }
               io_worker_pool_.PushSpillWorker(io_worker);
-              if (!status.ok()) {
-                for (const auto &object_id : objects_to_spill) {
+              size_t num_objects_spilled = r.spilled_objects_url_size();
+              RAY_CHECK(num_objects_spilled <= objects_to_spill.size());
```

Comment:
Can you comment that this assumes the spilling is always done in the right order? For example, when we spill A,B,C,D,E, this assumes A cannot fail when B succeeds iiuc right?

Reply:
That's correct. In the implementation, if there is any error, it'll just return. It assumes there is something broken and won't continue to write.